src/lib/shopping-analytics.js

```diff
@@ -0,0 +1,21 @@
+/* @flow */
+import type { FptiInput, Config } from '../types';
+
+import { trackFptiV2 } from './fpti';
+import { eventToFptiConverters, type EventToFptiInputMapping } from './shopping-event-conversions';
+
+const initEventPublisher = (config : Config) => {
+  return (converterToFpti : EventToFptiInputMapping) => {
+    return (event : Object) => {
+      const fptiInput : FptiInput = converterToFpti(event);
+      trackFptiV2(config, fptiInput);
+    };
+  };
+};
+
+export const setupTrackers = (config : Config) => {
+  const eventPublisher = initEventPublisher(config);
+  const converters = eventToFptiConverters(config);
+  const viewPage = eventPublisher(converters.viewPageToFpti);
+  return { viewPage };
```

Comment:
This might be very subjective, but I found the following to be better readable 

```suggestion
  const { viewPageToFpti: viewPageToFptiConverter } = eventToFptiConverters(config);
  
  return { viewPage: initEventPublisher(config)(viewPageToFptiConverter) };
};
```